src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
+ * input path_parameters.num_elements defines the number of constant-parameterization
+ * elements, and the returned path_parameters.num_elements defines the number of
+ * constant-period elements in the trajectory,
  *
- *  TODO: Remove when #1322 is merged.
+ * @param interpolation_period  [in] The duration of time between successive trajectory
+ * elements. In seconds.
  *
- * @param forwards_trajectory This is the trajectory that will be modified in place to
- * become a reverse trajectory
+ * @return A status indicating whether or not generation was successful
  */
-void static app_trajectory_planner_reversePositionTrajectoryDirection(
-    PositionTrajectory_t* forwards_trajectory);
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_generateConstantInterpolationPeriodPositionTrajectory(
+    PositionTrajectory_t* constant_period_trajectory,
+    FirmwareRobotPathParameters_t* path_parameters, float interpolation_period);
 
 /**
- * This function generates a velocity trajectory that corresponds to the time-optimal
- * velocity to follow a specified path. This profile is based on the input position
- * trajectory
- *
- * @param position_trajectory [in] The position trajectory to build the velocity
- * trajectory from
- * @param velocity_trajectory [out] The velocity trajectory that corresponds to the
- * time-optimal velocity to follow a specified path
+ * Function generates a velocity trajectory composed of X/Y/angular velocities and their
+ * time dependence.
+ *
+ * @param position_trajectory [in] A completely defined position trajectory.
+ *
+ * @param velocity_trajectory [out] A completely defined velocity trajectory that has
+ * equally time-spaced elements. The velocity trajectory has the name number of elements
+ * as the input position trajectory.
+ *
+ * @param num_elements [in] The number of elements in the trajectory.
  */
 void app_trajectory_planner_generateVelocityTrajectory(
-    PositionTrajectory_t* position_trajectory, VelocityTrajectory_t* velocity_trajectory);
+    PositionTrajectory_t* position_trajectory, VelocityTrajectory_t* velocity_trajectory,
+    unsigned int num_elements);
+
+/**
+ * Function modifies the value of the argument final speed so that the time duration
+ * between the initial and final speeds is equal to the parameter duration. This is done
+ * assuming constant acceleration over the given displacement.
+ *
+ * @param initial_speed [in] The initial speed at the start of a segment. In m/s.
+ *
+ * @param final_speed [in/out] A pointer to the final speed at the end of the segment. In
+ * m/s.
+ *
+ * @param duration [in] The duration of time between the initial and final speed points.
+ * This is also the time required to traverse the displacement parameter. In seconds.
+ *
+ * @param displacement [in] The distance between the initial and final velocity points. In
+ * meters.
+ */
+void app_trajectory_planner_modifySpeedToMatchDuration(float initial_speed,
```

Comment:
why not return the final speed?

Reply:
No particular reason other than I thought it made things cleaner to modify the value in the function. 

If you're opposed to this I'll change it 👍

Reply:
yes I prefer returning values if possible